### PR TITLE
Allow for custom IAM role annotation names.

### LINKED ISSUE #2963: Support for custom IAM annotation name
The annotation name used for kube2iam is currently hard-coded.

We use a custom name, and would like a flag to support setting this using `pachctl deploy`.

Patch in a moment.

### PR DESCRIPTION
**Pull request by jkinkead**
Fixes #2963 .

### CODE REVIEW

**Comment by msteffen** [comment]
Hi Jesse, I haven't dug into it yet, but when I tried to copy this change into an internal branch to run our test suite, it failed and emitted this error:
```
pachctl deploy local --no-guaranteed -d --dry-run  | kubectl  apply -f -
serviceaccount "pachyderm" created
clusterrole "pachyderm" created
clusterrolebinding "pachyderm" created
deployment "etcd" created
service "etcd" created
service "pachd" created
service "dash" created
deployment "dash" created
secret "pachyderm-storage-secret" created
The Deployment "pachd" is invalid: 
* spec.template.annotations: Invalid value: "": name part must be non-empty
* spec.template.annotations: Invalid value: "": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')
make: *** [launch-dev] Error 1
```

**Comment by jkinkead** [comment]
I've added a commit to this branch to pass the annotation name through to other places it's needed.

**Comment by msteffen** [comment]
Hi Jesse, sorry I let this languish for so long! On Friday I merged this into the test branch for this PR and unfortunately our internal CI failed because pachd couldn't come up; kubernetes gave this error:
```
The Deployment "pachd" is invalid: 
* spec.template.annotations: Invalid value: "": name part must be non-empty
* spec.template.annotations: Invalid value: "": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')
make: *** [launch-dev] Error 1
```
I wanted to let you know just to give you a status update, and as far as next steps:
- I can take a look at the PR and try to make sure no annotation is added if the field name is empty, though I have a few other projects I'm balancing at the moment
- Basically, while some of our tests won't pass in this branch, because they test enterprise features and external PRs can't download our test enterprise key, you should be able to use these tests to get all non-enterprise tests passing here. In other words, if you're understandably tired of the slow feedback loop this with PR, another option to get unblocked is to get to the point where all non-enterprise tests pass, which will make the likelihood that our enterprise tests pass we get this submitted much higher.

**Comment by jkinkead** [comment]
What's the best way to run the non-enterprise tests? Or, similarly, what set of tests should complete successfully?

**Comment by msteffen** [comment]
Ah, sorry, to clarify: under the comments on this page, there should be a box saying "All checks have failed" and then "continuous-integration/travis-ci/pr ... **Details**". Click "Details" and you'll see how far Travis got through our test suite.

Right now, all four test shards are failing, and if you click the first one one you're presented with a log from running the test. At the bottom of the log is:
```
The Deployment "pachd" is invalid: 
* spec.template.annotations: Invalid value: "": name part must be non-empty
* spec.template.annotations: Invalid value: "": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')
make: *** [launch-dev] Error 1
```
showing the problem.

I think the last three shards of our test suite should be able to pass, and in the first shard, we'll know we hit an enterprise test problem because the logs will end with some kind of error message about failing to fetch the enterprise key.

I hope this helps, and let me know if you have any other questions or anything!

**Comment by msteffen** [comment]
Also, if you want to test locally, you'll have to install minikube, but the following commands, run inside the pachyderm repo, are a large subset of our overall test suite:
```
make launch-dev-vm
go test -v ./src/server
```

**Comment by jkinkead** [comment]
Thanks, that should help!

**Comment by JoeyZwicker** [comment]
Hey @jkinkead and @msteffen Just checking in on the status of this PR

**Comment by jdoliner** [comment]
Closing for oldness feel free to reopen if we can fix the issues that prevent pachd from coming up.
